OpenRA.Mods.Common/Traits/Conditions/GrantChargeDrainCondition.cs

```diff
@@ -0,0 +1,216 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Activities;
+using OpenRA.Mods.Common.Orders;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Grants a condition for a certain time when deployed. Can be reversed anytime " +
+		"by undeploying. Leftover charges are taken into account when recharging.")]
+	public class GrantChargeDrainConditionInfo : PausableConditionalTraitInfo
+	{
+		[GrantedConditionReference]
+		[FieldLoader.Require]
+		[Desc("The condition granted after deploying.")]
+		public readonly string DeployedCondition = null;
+
+		[Desc("Time it takes to fully charge up again.")]
+		public readonly int ChargeTicks = 1000;
+
+		[Desc("The deployed state's length in ticks.")]
+		public readonly int DrainTicks = 100;
+
+		[Desc("Cursor to display when able to (un)deploy the actor.")]
+		public readonly string DeployCursor = "deploy";
+
+		[Desc("Cursor to display when unable to (un)deploy the actor.")]
+		public readonly string DeployBlockedCursor = "deploy-blocked";
+
+		public readonly bool StartsFullyCharged = false;
+
+		[VoiceReference]
+		public readonly string Voice = "Action";
+
+		public readonly bool ShowSelectionBar = true;
+		public readonly Color ChargingColor = Color.DarkRed;
+		public readonly Color DischargingColor = Color.DarkMagenta;
+
+		public override object Create(ActorInitializer init) { return new GrantChargeDrainCondition(init.Self, this); }
+	}
+
+	public enum TimedDeployState { Charging, Ready, Active, Deploying, Undeploying }
+
+	public class GrantChargeDrainCondition : PausableConditionalTrait<GrantChargeDrainConditionInfo>,
+		IResolveOrder, IIssueOrder, ISelectionBar, IOrderVoice, ISync, ITick, IIssueDeployOrder
+	{
+		readonly Actor self;
+		readonly bool canTurn;
```

Comment:
This doesn't appear to be used for anything?

Reply:
Removed it.